nearcache/nearcache_it_test.go

```diff
@@ -0,0 +1,327 @@
+	testCases := []mapTestCase{
+		{
+			name: "Put",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				v, err := tcx.M.Put(ctx, i, i)
+				if err != nil {
+					tcx.T.Fatal(err)
+				}
+				require.Equal(tcx.T, i, v)
+			},
+		},
+		{
+			name: "PutWithMaxIdle",
+			f: func(ctx context.Context, tcx it.MapTestContext, i int64) {
+				v, err := tcx.M.PutWithMaxIdle(ctx, i, i, 10*time.Minute)
+				if err != nil {
+					tcx.T.Fatal(err)
+				}
+				require.Equal(tcx.T, i, v)
+			},
+		},
+	}
+	invalidationRunner(t, testCases)
+}
+
+func clientCacheNearCacheBasicSlowRunner(t *testing.T, f func(tcx *it.NearCacheTestContext, size int64, valueFmt string)) {
+	testCases := []struct {
+		inMemoryFmt   nearcache.InMemoryFormat
+		serializeKeys bool
+	}{
+		{inMemoryFmt: nearcache.InMemoryFormatBinary, serializeKeys: true},
+		{inMemoryFmt: nearcache.InMemoryFormatBinary, serializeKeys: false},
+		{inMemoryFmt: nearcache.InMemoryFormatObject, serializeKeys: true},
+		{inMemoryFmt: nearcache.InMemoryFormatObject, serializeKeys: false},
+	}
+	for _, tc := range testCases {
+		testName := fmt.Sprintf("inMemoryFmt %s serializeKeys %t", inMemoryFmtToString(tc.inMemoryFmt), tc.serializeKeys)
+		t.Run(testName, func(t *testing.T) {
+			ncc := nearcache.Config{
+				Name:           "*",
+				InMemoryFormat: tc.inMemoryFmt,
+				SerializeKeys:  tc.serializeKeys,
+			}
+			ncc.SetInvalidateOnChange(false)
+			configCB := func(tcx it.MapTestContext) {
+				tcx.Config.AddNearCache(ncc)
+			}
+			valueFmt := "value-%d"
+			mtcx := &it.MapTestContext{
+				T:              t,
+				ConfigCallback: configCB,
+			}
+			mtcx.Tester(func(mtcx it.MapTestContext) {
+				nca := hz.MakeNearCacheAdapterFromMap(mtcx.M)
+				ci := hz.NewClientInternal(mtcx.Client)
+				tcx := it.NewNearCacheTestContext(mtcx.T, nca.(it.NearCacheAdapter), mtcx.M, &ncc, ci.SerializationService())
+				// assert that the Near Cache is empty
+				tcx.PopulateNearCacheDataAdapter(nearCacheDefaultRecordCount, valueFmt)
+				tcx.RequireNearCacheSize(0)
+				tcx.AssertNearCacheStats(0, 0, 0)
+				// populate the Near Cache
+				f(tcx, nearCacheDefaultRecordCount, valueFmt)
+				tcx.RequireNearCacheSize(nearCacheDefaultRecordCount)
+				tcx.AssertNearCacheStats(nearCacheDefaultRecordCount, 0, nearCacheDefaultRecordCount)
+				// generate Near Cache hits
+				f(tcx, nearCacheDefaultRecordCount, valueFmt)
+				tcx.RequireNearCacheSize(nearCacheDefaultRecordCount)
+				tcx.AssertNearCacheStats(nearCacheDefaultRecordCount, nearCacheDefaultRecordCount, nearCacheDefaultRecordCount)
+				tcx.AssertNearCacheContent(nearCacheDefaultRecordCount, valueFmt)
+				// TODO: assertNearCacheReferences
+			})
+		})
+	}
+}
+
+func invalidationRunner(t *testing.T, testCases []mapTestCase) {
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tcx := newNearCacheMapTestContext(t, nearcache.InMemoryFormatBinary, false)
```

Comment:
This should be true I guess. see [this](https://github.com/hazelcast/hazelcast/blob/14ef0dbefa4a6fa00a50a8873aeea02ec6c3f5e4/hazelcast/src/test/java/com/hazelcast/client/map/impl/nearcache/ClientMapNearCacheTest.java#L185). How does the test pass? 🤔

Reply:
Right. But Java client test also passes with `.setInvalidateOnChange(false)`.

Reply:
Fixed at: https://github.com/hazelcast/hazelcast-go-client/pull/806/commits/0340cb385720900db303b5811060b367f99c9526